components/org.wso2.carbon.identity.keyrotation/src/main/java/org.wso2.carbon.identity.keyrotation/dao/RegistryDAO.java

```diff
@@ -0,0 +1,179 @@
+    private static final String REG_ID = "REG_ID";
+    private static final String REG_NAME = "REG_NAME";
+    private static final String REG_VALUE = "REG_VALUE";
+    private static final String REG_TENANT_ID = "REG_TENANT_ID";
+    public static int updateCount = 0;
+    public static int failedUpdateCount = 0;
+
+    public RegistryDAO() {
+
+    }
+
+    public static RegistryDAO getInstance() {
+
+        return instance;
+    }
+
+    /**
+     * To retrieve the registry property data in REG_PROPERTY as chunks.
+     *
+     * @param startIndex        The start index of the data chunk.
+     * @param keyRotationConfig Configuration data needed to perform the task.
+     * @param property          Registry property value.
+     * @return List comprising of the records in the table.
+     * @throws KeyRotationException Exception thrown while retrieving data from REG_PROPERTY.
+     */
+    public List<RegistryProperty> getRegPropertyDataChunks(int startIndex, KeyRotationConfig keyRotationConfig,
+                                                           String property) throws KeyRotationException {
+
+        List<RegistryProperty> regPropertyList = new ArrayList<>();
+        String query = DBConstants.GET_REG_PROPERTY_DATA;
+        int firstIndex = startIndex;
+        int secIndex = DBConstants.CHUNK_SIZE;
+        try (Connection connection = DriverManager
+                .getConnection(keyRotationConfig.getNewRegDBUrl(), keyRotationConfig.getNewRegUsername(),
+                        keyRotationConfig.getNewRegPassword())) {
+            if (connection.getMetaData().getDriverName().contains(DBConstants.POSTGRESQL)) {
+                query = DBConstants.GET_REG_PROPERTY_DATA_POSTGRE;
+                firstIndex = DBConstants.CHUNK_SIZE;
+                secIndex = startIndex;
+            } else if (connection.getMetaData().getDriverName().contains(DBConstants.MSSQL) ||
+                    connection.getMetaData().getDriverName().contains(DBConstants.ORACLE)) {
+                query = DBConstants.GET_REG_PROPERTY_DATA_OTHER;
+            }
+            try (PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+                preparedStatement.setString(1, property);
+                preparedStatement.setInt(2, firstIndex);
+                preparedStatement.setInt(3, secIndex);
+                ResultSet resultSet = preparedStatement.executeQuery();
+                while (resultSet.next()) {
+                    regPropertyList
+                            .add(new RegistryProperty(resultSet.getString(REG_ID),
+                                    resultSet.getString(REG_NAME),
+                                    resultSet.getString(REG_VALUE),
+                                    resultSet.getString(REG_TENANT_ID)));
+                }
+            } catch (SQLException e) {
+                throw new KeyRotationException("Error while retrieving registry property " + property + " from " +
```

Comment:
throw new KeyRotationException("Error while retrieving registry property: " + property + " from " +
                        "REG_PROPERTY.", e);


notice the ":"

Reply:
Addressed with https://github.com/wso2/identity-tools/pull/13/commits/2c709c0658bfeb105b1836765c424d5ac02d18f5